extension/src/experiments/commands/index.ts

```diff
@@ -56,23 +60,39 @@ export const getPushExperimentCommand =
     }
 
     return Toast.showProgress('exp push', async progress => {
+      const repository = experiments.getRepository(dvcRoot)
+
+      const updateOnCompletion = () => {
+        repository.unsetPushing(ids)
+        return repository.update()
+      }
+
       progress.report({ increment: 0 })
 
       progress.report({ increment: 25, message: `Pushing ${ids.join(' ')}...` })
 
       const remainingProgress = 75
 
-      const stdout = await internalCommands.executeCommand(
-        AvailableCommands.EXP_PUSH,
-        dvcRoot,
-        ...ids
-      )
+      repository.setPushing(ids)
+
+      try {
+        const stdout = await internalCommands.executeCommand(
+          AvailableCommands.EXP_PUSH,
+          dvcRoot,
+          ...ids
+        )
+
+        void updateOnCompletion()
 
-      progress.report({
-        increment: remainingProgress,
-        message: convertUrlTextToLink(stdout)
-      })
+        progress.report({
+          increment: remainingProgress,
+          message: convertUrlTextToLink(stdout)
+        })
 
-      return Toast.delayProgressClosing(15000)
+        return Toast.delayProgressClosing(15000)
+      } catch (error: unknown) {
+        void updateOnCompletion()
+        throw error
```

Comment:
[F] This will be caught by `InternalCommands` and offer to show the output channel to the user (which will show the underlying error)